csrc/host_ir/host_ir.h

```diff
@@ -114,6 +114,48 @@ class PostOnStream : public Expr {
   }
 };
 
+class StreamIr : public Val {
```

Comment:
I feel the `Ir` in `StreamIr` is a bit redundant given it's already in namespace `hir`. But I'd totally understand if there's a particular naming convention you want to follow.